packit/config/aliases.py

```diff
@@ -230,45 +231,53 @@ def get_all_koji_targets() -> List[str]:
 @fallback_return_value(ALIASES)
 def get_aliases() -> Dict[str, List[str]]:
     """
-    Function to automatically determine fedora-all, fedora-stable, fedora-development,
-    fedora-latest and epel-all aliases.
+    Function to automatically determine fedora-* and epel-* aliases.
     Current data are fetched via bodhi client, with default base url
     `https://bodhi.fedoraproject.org/'.
 
     Returns:
         Dictionary containing aliases.
     """
-
     bodhi_client = get_bodhi_client()
     releases = bodhi_client.get_releases(exclude_archived=True)
-    aliases = defaultdict(list)
-    for release in releases.releases:
-
-        if release.id_prefix == "FEDORA" and release.name != "ELN":
-            name = release.long_name.lower().replace(" ", "-")
-            if release.state == "current":
-                aliases["fedora-stable"].append(name)
-            elif release.state == "pending":
-                aliases["fedora-development"].append(name)
-
-        elif release.id_prefix == "FEDORA-EPEL":
-            name = release.name.lower()
-            aliases["epel-all"].append(name)
-
-    if "fedora-development" in aliases:
-        aliases["fedora-development"].sort(key=lambda x: int(x.rsplit("-")[-1]))
-        # The Fedora with the highest version is "rawhide", but
-        # Bodhi always uses release names, and has no concept of "rawhide".
-        aliases["fedora-development"][-1] = "fedora-rawhide"
+    fedora_releases, epel_releases = [], []
 
-    if "fedora-stable" in aliases:
-        aliases["fedora-stable"].sort(key=lambda x: int(x.rsplit("-")[-1]))
-
-    aliases["fedora-all"] = aliases["fedora-stable"] + aliases["fedora-development"]
-
-    # fedora-rawhide is the last release, we want the second latest (the latest
-    # non rawhide release)
-    if len(aliases["fedora-all"]) >= 2:
-        aliases["fedora-latest"] = [aliases["fedora-all"][-2]]
-
-    return aliases
+    for release in releases.releases:
+        if release.state in ["current", "pending"]:
+            if release.id_prefix == "FEDORA" and release.name != "ELN":
+                name = release.long_name.lower().replace(" ", "-")
+                fedora_releases.append(name)
+            elif release.id_prefix == "FEDORA-EPEL":
+                name = release.name.lower()
+                epel_releases.append(name)
+
+    fedora_releases.sort(key=lambda x: int(x.rsplit("-")[-1]))
+    fedora_releases[-1] = "fedora-rawhide"
+
+    #  fedora-34   fedora-35   [ fedora-36 ]   fedora-rawhide
+    # <--------------------- all ---------------------------->
+    # <------ stable ------->
+    #                         <-------- development --------->
+    #                         <--- latest -->
+    #           <latest-stable>
+    # <----------- branched ---------------->
+    pending_releases = len(fedora_releases) - 2
```

Comment:
Oh, wait, if this is so then this needs some more fixing :D

Reply:
Now it should be fixed. Instead of relying on the number of releases, "pending" and "current" releases are clearly separated, sorted and all aliases are composed of these two, which makes things much clearer.

I've also added a test for the case above: after a release and before EOL there are 3 current releases and 1 pending (rawhide).